pkg/controller/statusmanager/pod_status.go

```diff
@@ -84,53 +84,93 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
 		*conditions = append(*conditions, newCondition)
 		return true
 	}
-
-	if existingCondition.Status != newCondition.Status {
+	if (existingCondition.Status != newCondition.Status || existingCondition.Reason != newCondition.Reason ||
+		existingCondition.Message != newCondition.Message) {
+		changed = true
```

Comment:
`changed = true` is not needed

Reply:
it should be returned at the end of function?

Reply:
Got it